src/main/java/duke/pack/Deadline.java

```diff
@@ -0,0 +1,48 @@
+package duke.pack;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected String time;
+    protected LocalDate date;
+    protected String fullDesc;
+
+    /**
+     * creates a deadline type of task
+     * @param description the task to be done
+     * @param time the time the task is due
+     * @param date the date the task is due
+     * @param fullDesc the full description of task
+     */
+    public Deadline(String description, String time, LocalDate date, String fullDesc) {
+        super(description, fullDesc);
+        this.time = time;
+        this.date = date;
+    }
+
+    /**
+     * Formats the task to be saved in hard disk
+     * @return a string of the formatted task
+     */
+    @Override
+    public String formatForFile() {
+        String type = "D";
```

Comment:
The same goes for constant pieces of information that you have used in this class. It is always a good practice to define them as constant variables.